_posts/2018-02-15-totw-88.md

```diff
@@ -0,0 +1,152 @@
+---
+title: "Tip of the Week #88: Initialization: =, (), and {}"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/88
+type: markdown
+order: "088"
+---
+
+Originally posted as TotW #88 on Jan 27, 2015
+
+*by Titus Winters (who/titus), on behalf of the C++ Style Arbiters*
+
+C++11 provided a new syntax referred to as "uniform initialization syntax" that
+was supposed to unify all of the various styles of initialization, avoid the
+[Most Vexing Parse](http://en.wikipedia.org/wiki/Most_vexing_parse), and avoid
+narrowing conversions. This new mechanism means we now have
+[yet another](https://xkcd.com/927/) syntax for initialization, with its own
+tradeoffs.
+
+## C++11 Brace Initialization
+
+Some uniform initialization syntax proponents would suggest that we use {}s and
+direct initialization (no use of the ‘=’, although in most cases both forms call
+the same constructor) for initialization of all types:
+
+```c++
+int x{2};
+string foo{"Hello World"};
+std::vector<int> v{1, 2, 3};
+```
+
+vs. (for instance):
+
+```c++
+int x = 2;
+string foo = "Hello World";
+std::vector<int> v = {1, 2, 3};
+```
+
+This approach has two shortcomings. First, "uniform" is a stretch: there are
+cases where ambiguity still exists (for the casual reader, not the compiler) in
+what is being called and how.
+
+```c++
+std::vector<string> strings{2}; // A vector of two empty strings.
+std::vector<int> ints{2};       // A vector containing only the integer 2.
+```
+
+Second: this syntax is not exactly intuitive: no other common language uses
+something like it. The language can certainly introduce new and surprising
+syntax, and there are technical reasons why it’s necessary in some cases. The
```

Comment:
necessary in some cases - *especially* in generic code.

Reply:
Done in https://github.com/abseil/abseil.github.io/pull/145